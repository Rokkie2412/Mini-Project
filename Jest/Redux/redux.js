export const ADD = 'ADD'
export const DELETE = 'DELETE'

export const addArray = (data) =>({
    type:ADD,
    payload:data,
})

export const deleteFromArray = (title) => ({
    type:DELETE,
    payload:title,
})

let initialState = {
    arr:[]
}

export const mainReducer = (state=initialState,action) => {
    switch(action.type){
        case ADD:
            return{
                ...state,arr:[...state.arr,action.payload]
            }
        case DELETE:
            return{
                ...state,arr:state.arr.filter(arr => arr.id !== action.payload)
            }
        default:
            return state
    }
}

export default mainReducer