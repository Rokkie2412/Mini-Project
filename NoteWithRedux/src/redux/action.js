import { ADDARRAY } from "./actionType";

export const addArray = (data) => ({
    type:ADDARRAY,
    payload:data
})