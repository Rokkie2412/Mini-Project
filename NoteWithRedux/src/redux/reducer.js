import { ADDARRAY } from "./actionType";

const initialState = {
    arr:[],
    
};

export const mainReducer = (state = initialState,action) => {
    switch (action.type) {
        case ADDARRAY:
            return{
                ...state,arr:[...state.arr,action.payload]
            }
           
        default:
            return state;
    }
}