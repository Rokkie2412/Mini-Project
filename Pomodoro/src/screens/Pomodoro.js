import React,{useState,useEffect} from 'react';
import { View,Text,Pressable,KeyboardAvoidingView,ToastAndroid,Vibration} from 'react-native';
import styles from '../styles/Pomodorostyles'
import {displayTimer,startTimer} from '../functions/PomodoroFunction'
import Ion from 'react-native-vector-icons/Ionicons'
import Settings from '../components/SetttingsModal';
import { useSelector } from 'react-redux';
import reactNativeBackgroundTimer from 'react-native-background-timer';


const Pomodoro = () =>{
    const [title,setTitle] = useState('Work Time!')
    const workhourinsec = useSelector(state=>state.resultWorkinSecond)
    const Breakinsec = useSelector(state =>state.resultBreakinSecond)
    const longBreakinSec = useSelector(state=>state.resultLongBreakinSecond)
    const [secondLeft,setSecondLeft] = useState(workhourinsec);
    const [timerOn, setTimerOn] = useState(false)
    const [modalShown,setModalShown] = useState(false)
    const [count,setCount] = useState(1)

    useEffect(()=>{ 
    if (timerOn === true){
        startTimer(setSecondLeft,workhourinsec,Breakinsec,longBreakinSec,count,setCount)
    } 
    else reactNativeBackgroundTimer.stopBackgroundTimer()

    return()=>{
      reactNativeBackgroundTimer.stopBackgroundTimer()
    }
  },[timerOn])

    useEffect(()=>{
    if (secondLeft === 0) {
        if (count%2 === 0) {
            setSecondLeft(workhourinsec)
            setCount(count+1)
            setTitle("Keep it Up!")
            Vibration.vibrate(1000)
        }else if (count%2 === 1) {
            if (count <= 6) {
                setSecondLeft(Breakinsec)
                setCount(count+1)
                setTitle("Short Break! Catch a Breath")
                Vibration.vibrate(1000)
            }else if (count === 7) {
                setSecondLeft(longBreakinSec)
                setCount(0)
                setTitle("Break Time! Enjoy Your Break")
                Vibration.vibrate(1500)
            }
        }
        // reactNativeBackgroundTimer.stopBackgroundTimer()
    }
  },[secondLeft])

    return(
        <KeyboardAvoidingView style={styles.mainContainer}>
            <Text style={styles.Header}>Pomodoro Timer</Text>
            <View style={styles.TimerSection}>
                <View style={styles.timeSection}>
                    <Text style={styles.showsection}>{title}</Text>
                    <Text style={styles.time}>{displayTimer(secondLeft).displayHours} : {displayTimer(secondLeft).displayMins} : {displayTimer(secondLeft).displaySecs}</Text>
                </View>
                <View style={styles.iconSection}>
                    {timerOn === false?
                    <Pressable onPress={()=>setTimerOn(true)}>
                        <Ion name='play' style={styles.icon}/>
                    </Pressable>:
                    <Pressable onPress={()=>setTimerOn(false)}>
                        <Ion name='pause' style={styles.icon}/>
                    </Pressable>
                    }
                    <Pressable
                    onPress={()=>{
                        reactNativeBackgroundTimer.stopBackgroundTimer()
                        setSecondLeft(workhourinsec)
                        setCount(1)
                        setTimerOn(false)
                        setTitle('Timer Stop')
                    }}
                    >
                        <Ion name='stop' style={styles.icon}/>
                    </Pressable>
                </View>
            </View>
            <View style={styles.buttonSection}>
                <Pressable onPress={()=>{
                    {timerOn === true ? ToastAndroid.show("Timer is on!",ToastAndroid.LONG):
                    setModalShown(!modalShown)
                    }
                }}>
                    <Text style={styles.button}>Settings <Ion name='settings' size={18}/></Text>
                </Pressable>
            </View>
            <Settings show={modalShown} setshow={setModalShown} setwork={setSecondLeft}/>    
        </KeyboardAvoidingView>
    )
}

export default Pomodoro