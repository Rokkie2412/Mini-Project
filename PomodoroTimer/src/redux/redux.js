
/**
 * @format
 * @flow
 */
export const EXCHAHNGETOSECONDS : string = 'EXCHANGETOSECOND'
export const EXCHAHNGETOSECONDBREAK :string = 'EXCHANGETOSECONDBREAK'
export const EXCHAHNGETOSECONDLONGBREAK :string = 'EXCHANGETOSECONDLONGBREAK'
export const GETDATA:string = 'GETDATA'

export const exchangetosecwork = (hour:string,minute:string,second:string):Object =>({
    type:EXCHAHNGETOSECONDS,
    payloadhourwork : hour,
    payloadminutework : minute,
    payloadsecondwork : second
})

export const exchangetosecBreak = (minute:string,second:string):Object =>({
    type:EXCHAHNGETOSECONDBREAK,
    payloadminutebreak : minute,
    payloadsecondbreak : second
})

export const exchangetosecLongBreak = (minute:string,second:string):Object =>({
    type:EXCHAHNGETOSECONDLONGBREAK,
    payloadminutelongbreak : minute,
    payloadsecondlongbreak : second
})

let initialState : Object  ={
    resultWorkinSecond : 0,
    resultBreakinSecond : 0,
    resultLongBreakinSecond : 0
}

export const mainReducer = (state:Object=initialState, action:function):Function =>{
    switch (action.type) {
        case EXCHAHNGETOSECONDS:
            
            const hourTosecond : number = action.payloadhourwork * 3600
            const minuteTosecond : number = action.payloadminutework * 60
            const res : number = hourTosecond+minuteTosecond+action.payloadsecondwork
            return {
                ...state,
                resultWorkinSecond:res
            }
        case EXCHAHNGETOSECONDBREAK:
            const minuteTosecondBreak : number= action.payloadminutebreak * 60
            const resBreak : number = minuteTosecondBreak+action.payloadsecondbreak
            return {
                ...state,
                resultBreakinSecond:resBreak
            }
        case EXCHAHNGETOSECONDLONGBREAK:
            const minuteToSecondLongBreak : number= action.payloadminutelongbreak * 60
            const resLongBreak : number = minuteToSecondLongBreak+action.payloadsecondlongbreak
            return{
                ...state,
                resultLongBreakinSecond:resLongBreak
            }
        default:
            return state
    }
}

export default mainReducer