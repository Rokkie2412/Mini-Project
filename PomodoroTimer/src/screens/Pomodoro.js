/**
 * @format
 * @flow
 */

import * as React from 'react';
import {useState, useEffect} from 'react';
import {
  View,
  Text,
  Pressable,
  KeyboardAvoidingView,
  ToastAndroid,
  Vibration,
} from 'react-native';
import styles from '../styles/Pomodorostyles';
import {
  displayTimer,
  startTimer,
  ChannelTest,
  handleNotification,
  handleNotificationLongBreak,
  timerset,
  ButtonDisable

} from '../functions/PomodoroFunction';
import Ion from 'react-native-vector-icons/Ionicons';
import Settings from '../components/SetttingsModal';
import Help from '../components/HelpModal';
import {useSelector} from 'react-redux';
import reactNativeBackgroundTimer from 'react-native-background-timer';

const Pomodoro = (): React.Node => {
  const [title, setTitle] = useState<string>('Work Time!');
  const workhourinsec: number = useSelector(state => state.resultWorkinSecond);
  const Breakinsec: number = useSelector(state => state.resultBreakinSecond);
  const longBreakinSec: number = useSelector(
    state => state.resultLongBreakinSecond,
  );
  const [secondLeft, setSecondLeft] = useState<number>(workhourinsec);
  const [timerOn, setTimerOn] = useState<boolean>(false);
  const [modalShown, setModalShown] = useState<boolean>(false);
  const [count, setCount] = useState<number>(1);
  const [judul, setJudul] = useState<string>('Timer to work!');
  const [pesan, setPesan] = useState<string>(
    'Time to work, keep it up! u doing good!',
  );
  const [wokrCycle, setWorkCycle] = useState<number>(0);
  const [help, setHelp] = useState<boolean>(false);

  useEffect(() => {
    ChannelTest();
  }, []);

  useEffect(() => {
    if (timerOn === true) {
      startTimer(setSecondLeft);
    } else reactNativeBackgroundTimer.stopBackgroundTimer();

    return () => {
      reactNativeBackgroundTimer.stopBackgroundTimer();
    };
  }, [timerOn]);

  useEffect(()=>{
    setSecondLeft(workhourinsec)
  },[workhourinsec])

  useEffect(() => {
    timerset(count,setJudul,setPesan,setSecondLeft,workhourinsec,setCount,setTitle,judul,pesan,Breakinsec,longBreakinSec,setWorkCycle,wokrCycle,secondLeft)
  }, [secondLeft,setSecondLeft]);
  return (
    <KeyboardAvoidingView style={styles.mainContainer}>
      <View style={styles.headerContainer}>
        <Text style={styles.Header}>Pomodoro Timer</Text>
      </View>
      <Pressable 
      testID='Help'
      onPress={() => setHelp(true)}>
        <Ion name="help-circle-outline" style={styles.HelpIcon} />
      </Pressable>
      <View style={styles.TimerSection}>
        <View style={styles.timeSection}>
          <Text style={styles.showsection}>{title}</Text>
          <Text style={styles.time}>
            {displayTimer(secondLeft).displayHours} :{' '}
            {displayTimer(secondLeft).displayMins} :{' '}
            {displayTimer(secondLeft).displaySecs}
          </Text>
        </View>

        <View style={styles.iconSection}>
          {timerOn === false ? (
            <Pressable
              testID='play'
              onPress={() => {
                setTimerOn(true);
                setTitle('Timer Play');
              }}>
              <Ion name="play" style={styles.icon} />
            </Pressable>
          ) : (
            <Pressable
            testID='pause'
              onPress={() => {
                setTimerOn(false);
                setTitle('Timer Pause');
              }}>
              <Ion name="pause" style={styles.icon} />
            </Pressable>
          )}
          <Pressable
            testID='stop'
            onPress={() => {
              reactNativeBackgroundTimer.stopBackgroundTimer();
              setSecondLeft(workhourinsec);
              setCount(1);
              setTimerOn(false);
              setTitle('Timer Stop');
              setWorkCycle(0);
            }}>
            <Ion name="stop" style={styles.icon} />
          </Pressable>
        </View>
        <View>
          <Text style={styles.cycle}>
            Work cycle has been done :{' '}
            <Text style={{color: '#00ADB5'}}>{wokrCycle}</Text>
          </Text>
        </View>
      </View>

      <View style={styles.buttonSection}>
        <Pressable
        testID='setting'
          onPress={() => {
            ButtonDisable(timerOn,setModalShown,modalShown)
          }}>
          <Text style={styles.button}>
            Settings <Ion name="settings" size={18} />
          </Text>
        </Pressable>
      </View>
      <Settings
        show={modalShown}
        setshow={setModalShown}
        setwork={setSecondLeft}
      />
      <Help HelpVisible={help} setHelpVisible={setHelp} />
    </KeyboardAvoidingView>
  );
};

export default Pomodoro;
