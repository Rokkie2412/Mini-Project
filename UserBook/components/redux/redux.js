export const ADD_CONTACT = 'ADD_CONTACT'
export const SHOW_CONTACT = 'SHOW_CONTACT'
export const EDITCONTACT = 'EDITCONTACT'

const ApiUrl = 'https://simple-contact-crud.herokuapp.com/contact'


export const addContact = (firstName,lastName,age,image) =>({
    type:ADD_CONTACT,
    payloadFirstname : firstName,
    payloadLastName : lastName,
    payAge : age,
    payloadImage : image
})

export const editContact = (itemId,firstName,lastName,age,image) => ({
    type:EDITCONTACT,
    payloadID : itemId,
    payloadFirstname : firstName,
    payloadLastName : lastName,
    payAge : age,
    payloadImage : image
})

let initialState = {
    arr:[],
}

export const mainReducer = (state=initialState, action) => {
    switch (action.type) {
        case ADD_CONTACT:
            const postContact = () =>{
                fetch('https://simple-contact-crud.herokuapp.com/contact',{
                    method:'POST',
                    body: JSON.stringify({
                        firstName : action.payloadFirstname,
                        lastName : action.payloadLastName,
                        age : action.payAge,
                        photo : action.payloadImage
                    }),
                    headers:{
                        'Content-Type' : 'application/json',
                    }
                }).then(Response=>Response.json()).then(json=>console.log(json))
            }
            return(postContact())
        case EDITCONTACT :
            const editContact = () =>{
                const tempid = action.payloadID 
                
                fetch(`https://simple-contact-crud.herokuapp.com/contact/${tempid}`,{
                    method:'PUT',
                    body: JSON.stringify({
                        firstName : action.payloadFirstname,
                        lastName : action.payloadLastName,
                        age : action.payAge,
                        photo : action.payloadImage
                    }),
                    headers:{
                        'Content-Type' : 'application/json',
                    }
                 }).then(Response=>Response.json()).then(json=>console.log(json))
            }
            return(editContact())
        default:
            return state
    }
}