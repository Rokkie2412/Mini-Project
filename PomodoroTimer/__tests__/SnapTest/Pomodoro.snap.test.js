import React from 'react';
import {create,act} from 'react-test-renderer';
import Pomodoro from '../../src/screens/Pomodoro';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import {mainReducer} from '../../src/redux/redux';

jest.mock('react-native-vector-icons/Ionicons', () => 'MockedIonicons');

jest.mock('react-native-background-timer',() =>{
  return{
    runBackgroundTimer: jest.fn(),
    stopBackgroundTimer : jest.fn()
  }
})

jest.mock('react-native-push-notification', () => {
  return {
    configure: jest.fn(),
    createChannel: jest.fn(),
    onRegister: jest.fn(),
    onNotification: jest.fn(),
    addEventListener: jest.fn(),
    requestPermissions: jest.fn(() => Promise.resolve),
    localNotification:jest.fn()
  };
});

jest.useFakeTimers();
const store = createStore(mainReducer, {status: 'default status'});
const tree = create(
  <Provider store={store}>
    <Pomodoro />
  </Provider>,
);

describe('Render Pomodoro Screen',()=>{
  describe('Snapshot testing for pomodor screen',()=>{
    it('Render correctly for pomodoro screen', () => {
      expect(tree).toMatchSnapshot();
    });
  })
  describe('Setting Button act',()=>{
    const setting = tree.root.findByProps({testID:'setting'}).props
    act(()=>setting.onPress())
  })
  describe('Setting Button act',()=>{
    const Help = tree.root.findByProps({testID:'Help'}).props
    act(()=>Help.onPress())
  })
  describe('Setting Button act',()=>{
    const play = tree.root.findByProps({testID:'play'}).props
    act(()=>play.onPress())
  })
  describe('Setting Button act',()=>{
    const pause = tree.root.findByProps({testID:'pause'}).props
    act(()=>pause.onPress())
  })
  describe('Setting Button act',()=>{
    const stop = tree.root.findByProps({testID:'stop'}).props
    act(()=>stop.onPress())
  })
})

