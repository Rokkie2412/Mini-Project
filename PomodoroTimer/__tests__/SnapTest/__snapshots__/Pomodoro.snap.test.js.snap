// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Pomodoro Screen Snapshot testing for pomodor screen Render correctly for pomodoro screen 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#222831",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#EEEEEE",
          "fontSize": 30,
          "fontWeight": "bold",
          "margin": 20,
          "position": "absolute",
          "textAlign": "center",
        }
      }
    >
      Pomodoro Timer
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="Help"
  >
    <MockedIonicons
      name="help-circle-outline"
      style={
        Object {
          "alignSelf": "flex-end",
          "color": "#00ADB5",
          "fontSize": 33,
          "fontWeight": "bold",
          "marginVertical": 15,
          "padding": 10,
          "textAlign": "center",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#00ADB5",
          "borderRadius": 30,
          "marginHorizontal": 15,
          "maxHeight": 600.3000000000001,
          "maxWidth": 712.5,
          "paddingVertical": 40,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#393E46",
            "fontSize": 30,
            "fontWeight": "bold",
            "marginTop": 20,
            "textAlign": "center",
          }
        }
      >
        Timer Stop
      </Text>
      <Text
        style={
          Object {
            "borderRadius": 30,
            "color": "#EEEEEE",
            "fontSize": 59,
            "fontWeight": "bold",
            "marginHorizontal": "5%",
            "paddingVertical": 60,
            "textAlign": "center",
          }
        }
      >
        NaN
         :
         
        NaN
         :
         
        NaN
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="play"
      >
        <MockedIonicons
          name="play"
          style={
            Object {
              "backgroundColor": "#393E46",
              "borderRadius": 50,
              "color": "#fff",
              "fontSize": 35,
              "margin": 10,
              "marginTop": 15,
              "padding": 24,
            }
          }
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="stop"
      >
        <MockedIonicons
          name="stop"
          style={
            Object {
              "backgroundColor": "#393E46",
              "borderRadius": 50,
              "color": "#fff",
              "fontSize": 35,
              "margin": 10,
              "marginTop": 15,
              "padding": 24,
            }
          }
        />
      </View>
    </View>
    <View>
      <Text
        style={
          Object {
            "color": "#EEEEEE",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginTop": 15,
            "textAlign": "center",
          }
        }
      >
        Work cycle has been done :
         
        <Text
          style={
            Object {
              "color": "#00ADB5",
            }
          }
        >
          0
        </Text>
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="setting"
    >
      <Text
        style={
          Object {
            "backgroundColor": "#393E46",
            "borderRadius": 50,
            "color": "#EEEEEE",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 10,
            "padding": 15,
          }
        }
      >
        Settings 
        <MockedIonicons
          name="settings"
          size={18}
        />
      </Text>
    </View>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "backgroundColor": "#393E46",
          "borderRadius": 30,
          "flex": 1,
          "height": 933.8,
          "margin": "auto",
          "marginHorizontal": "5%",
          "marginVertical": "27%",
          "width": 675,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#EEEEEE",
              "fontSize": 22,
              "fontWeight": "bold",
              "margin": 10,
            }
          }
        >
          Settings
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#222831",
            "borderRadius": 20,
            "marginHorizontal": 20,
            "marginVertical": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#EEEEEE",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginTop": 7,
              "textAlign": "center",
            }
          }
        >
          Set Work Timer
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <TextInput
            keyboardType="number-pad"
            onChangeText={[Function]}
            placeholder="00"
            placeholderTextColor="#EEEEEE"
            style={
              Object {
                "borderBottomColor": "#eeeeee",
                "borderBottomWidth": 2,
                "color": "#EEEEEE",
                "fontSize": 18,
                "margin": 10,
                "marginHorizontal": 8,
                "padding": 10,
                "textAlign": "center",
              }
            }
            testID="sethour"
            value=""
          />
          <TextInput
            keyboardType="number-pad"
            onChangeText={[Function]}
            placeholder="00"
            placeholderTextColor="#EEEEEE"
            style={
              Object {
                "borderBottomColor": "#eeeeee",
                "borderBottomWidth": 2,
                "color": "#EEEEEE",
                "fontSize": 18,
                "margin": 10,
                "marginHorizontal": 8,
                "padding": 10,
                "textAlign": "center",
              }
            }
            testID="setminute"
            value=""
          />
          <TextInput
            keyboardType="number-pad"
            onChangeText={[Function]}
            placeholder="00"
            placeholderTextColor="#EEEEEE"
            style={
              Object {
                "borderBottomColor": "#eeeeee",
                "borderBottomWidth": 2,
                "color": "#EEEEEE",
                "fontSize": 18,
                "margin": 10,
                "marginHorizontal": 8,
                "padding": 10,
                "textAlign": "center",
              }
            }
            testID="setsecond"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#222831",
            "borderRadius": 20,
            "marginHorizontal": 20,
            "marginVertical": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#EEEEEE",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginTop": 7,
              "textAlign": "center",
            }
          }
        >
          Set Short Break Timer
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <TextInput
            keyboardType="number-pad"
            onChangeText={[Function]}
            placeholder="00"
            placeholderTextColor="#EEEEEE"
            style={
              Object {
                "borderBottomColor": "#eeeeee",
                "borderBottomWidth": 2,
                "color": "#EEEEEE",
                "fontSize": 18,
                "margin": 10,
                "marginHorizontal": 8,
                "padding": 10,
                "textAlign": "center",
              }
            }
            testID="setshortminute"
            value=""
          />
          <TextInput
            keyboardType="number-pad"
            onChangeText={[Function]}
            placeholder="00"
            placeholderTextColor="#EEEEEE"
            style={
              Object {
                "borderBottomColor": "#eeeeee",
                "borderBottomWidth": 2,
                "color": "#EEEEEE",
                "fontSize": 18,
                "margin": 10,
                "marginHorizontal": 8,
                "padding": 10,
                "textAlign": "center",
              }
            }
            testID="setshortsecond"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#222831",
            "borderRadius": 20,
            "marginHorizontal": 20,
            "marginVertical": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#EEEEEE",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginTop": 7,
              "textAlign": "center",
            }
          }
        >
          Set Long Break Timer
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <TextInput
            keyboardType="number-pad"
            onChangeText={[Function]}
            placeholder="00"
            placeholderTextColor="#EEEEEE"
            style={
              Object {
                "borderBottomColor": "#eeeeee",
                "borderBottomWidth": 2,
                "color": "#EEEEEE",
                "fontSize": 18,
                "margin": 10,
                "marginHorizontal": 8,
                "padding": 10,
                "textAlign": "center",
              }
            }
            testID="setlongminute"
            value=""
          />
          <TextInput
            keyboardType="number-pad"
            onChangeText={[Function]}
            placeholder="00"
            placeholderTextColor="#EEEEEE"
            style={
              Object {
                "borderBottomColor": "#eeeeee",
                "borderBottomWidth": 2,
                "color": "#EEEEEE",
                "fontSize": 18,
                "margin": 10,
                "marginHorizontal": 8,
                "padding": 10,
                "textAlign": "center",
              }
            }
            testID="setlongsecond"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-evenly",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="CloseSetting"
        >
          <Text
            style={
              Object {
                "backgroundColor": "#00ADB5",
                "borderRadius": 50,
                "color": "#EEEEEE",
                "fontSize": 20,
                "fontWeight": "bold",
                "marginTop": 30,
                "padding": 15,
              }
            }
          >
            Cancel
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="SaveSetting"
        >
          <Text
            style={
              Object {
                "backgroundColor": "#00ADB5",
                "borderRadius": 50,
                "color": "#EEEEEE",
                "fontSize": 20,
                "fontWeight": "bold",
                "marginTop": 30,
                "padding": 15,
              }
            }
          >
            Save
          </Text>
        </View>
      </View>
    </View>
  </Modal>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "backgroundColor": "#393E46",
          "borderRadius": 30,
          "flex": 1,
          "height": "75%",
          "margin": "auto",
          "marginHorizontal": "5%",
          "marginVertical": "25%",
          "position": "absolute",
          "width": "90%",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="setupVisibleModal"
        >
          <MockedIonicons
            name="close"
            style={
              Object {
                "color": "#EEEE",
                "fontSize": 30,
                "margin": 10,
              }
            }
          />
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#EEEE",
            "fontSize": 30,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        How to use
      </Text>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#222831",
            "borderRadius": 25,
            "height": "71%",
            "margin": 12,
            "paddingHorizontal": 20,
            "width": "92%",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#EEEE",
                "fontSize": 17,
                "paddingLeft": 6,
                "paddingVertical": 17,
              }
            }
          >
            1.
          </Text>
          <Text
            style={
              Object {
                "color": "#EEEE",
                "fontSize": 16,
                "paddingHorizontal": 12,
                "paddingVertical": 17,
              }
            }
          >
            For better convenience, please lock the app on background
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#EEEE",
                "fontSize": 17,
                "paddingLeft": 6,
                "paddingVertical": 17,
              }
            }
          >
            2.
          </Text>
          <Text
            style={
              Object {
                "color": "#EEEE",
                "fontSize": 16,
                "paddingHorizontal": 12,
                "paddingVertical": 17,
              }
            }
          >
            Before use Doro Timer, you can set your timer that set for you in setting button
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#EEEE",
                "fontSize": 17,
                "paddingLeft": 6,
                "paddingVertical": 17,
              }
            }
          >
            3.
          </Text>
          <Text
            style={
              Object {
                "color": "#EEEE",
                "fontSize": 16,
                "paddingHorizontal": 12,
                "paddingVertical": 17,
              }
            }
          >
            To start timer, press play button, and to pause press pause button. Button will change every you press the button
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#EEEE",
                "fontSize": 17,
                "paddingLeft": 6,
                "paddingVertical": 17,
              }
            }
          >
            4.
          </Text>
          <Text
            style={
              Object {
                "color": "#EEEE",
                "fontSize": 16,
                "paddingHorizontal": 12,
                "paddingVertical": 17,
              }
            }
          >
            To stop timer, press stop button, timer will get reset to first condition
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#EEEE",
                "fontSize": 17,
                "paddingLeft": 6,
                "paddingVertical": 17,
              }
            }
          >
            5.
          </Text>
          <Text
            style={
              Object {
                "color": "#EEEE",
                "fontSize": 16,
                "paddingHorizontal": 12,
                "paddingVertical": 17,
              }
            }
          >
            When timer on you can't open setting, you must stop your timer first
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
        testID="VisibleModal"
      >
        <Text
          style={
            Object {
              "backgroundColor": "#00ADB5",
              "borderRadius": 50,
              "color": "#EEEE",
              "fontSize": 22,
              "fontWeight": "bold",
              "padding": 10,
            }
          }
        >
          Close
        </Text>
      </View>
    </View>
  </Modal>
</View>
`;
